### ISSUE TITLE
Backup page listing empty when network storage is configured

### ISSUE THREAD

**Issue by joshtbernstein** [home-assistant/supervisor#4357]
### The problem

I've added an CIFS/SMB share as a backup destination.

Server: hostname.example.com
Remote Share: share\folder

After doing so, the backup page no longer lists any backups, local or on the share. Creating a backup works (it's created on the share), but isn't listed.

### What version of Home Assistant Core has the issue?

core-2023.6.0

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Backup

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/backup/

### Anything in the logs that might be useful for us?

```txt
Supervisor:
23-06-08 12:29:37 INFO (MainThread) [supervisor.backups.manager] Found 38 backup files

Core:
2023-06-08 12:29:47.932 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /backups/reload request
2023-06-08 12:29:47.934 ERROR (MainThread) [homeassistant.components.hassio] Failed to to call /backups/reload -
```

**Comment by gawdzilla-dev** [comment]
I'm having a similar issue -- the backups page will no longer load at all after successfully adding a CIFS share for backups.  The loading spinner just spins -- doesn't even time out after 5 minutes.

I restarted HA entirely -- no change.
I deleted the CIFS share (no additional storage at all now) -- no change.
Restarted it again -- no change.

Backups page still won't load.

No errors in HA Core logs.  Supervisor logs only shows a bunch of failed attempts to contact Sentry.io (blocked by my piHole).  Nothing else in the other logs.

<details><summary>System Information</summary>

version | core-2023.6.1
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.3
os_name | Linux
os_version | 6.1.29
arch | x86_64
timezone | America/Los_Angeles
config_dir | /config

</details>


<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 10.2
-- | --
update_channel | stable
supervisor_version | supervisor-2023.06.2
agent_version | 1.5.1
docker_version | 23.0.6
disk_total | 30.8 GB
disk_used | 6.4 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.6.0), Log Viewer (0.15.1), Mosquitto broker (6.2.1), Terminal & SSH (9.7.1)

</details>

**Comment by perjury** [comment]
Same problem. 2023.7

**Comment by perjury** [comment]
I'm getting these errors in the log

Logger: homeassistant.components.hassio
Source: components/hassio/websocket_api.py:130
Integration: Home Assistant Supervisor (documentation, issues)
First occurred: 11 July 2023 at 21:56:57 (20 occurrences)
Last logged: 09:39:34

Failed to to call /resolution/info -
Failed to to call /os/info -
Failed to to call /addons -
Failed to to call /store -
Failed to to call /backups/reload -

and
Logger: homeassistant.components.hassio.handler
Source: components/hassio/handler.py:557
Integration: Home Assistant Supervisor (documentation, issues)
First occurred: 11 July 2023 at 21:56:57 (20 occurrences)
Last logged: 09:39:34

Timeout on /resolution/info request
Timeout on /os/info request
Timeout on /addons request
Timeout on /store request
Timeout on /backups/reload request

**Comment by perjury** [comment]
I have discovered that if I reduce the number of backup files then it works. It appears as if the request to list backup files times out

**Comment by circa1665** [comment]
Having the same issue:

Logger: homeassistant.components.hassio.handler
Source: components/hassio/handler.py:557
Integration: Home Assistant Supervisor ([documentation](https://www.home-assistant.io/integrations/hassio), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+hassio%22))
First occurred: 15:49:41 (4 occurrences)
Last logged: 15:50:36

Timeout on /backups/reload request

**Comment by PV-Joe** [comment]
Same problem here. The list of backup files in Home Assistant on my Fritz NAS is empty.

Before it took a while for the files to show up.

Please fix that bug. 🙏

**Comment by Yorkie71** [comment]
I believe this is a duplicate [issue here. ](https://github.com/home-assistant/supervisor/issues/4473)
I have worked around this issue by switching to NFS.

**Comment by romainrossi** [comment]
I have a _similar_ issue : the list is empty when the NAS is powered OFF; but the list is correctly displayed when it is ON.
This append even so I have LOCAL backups (on the HomeAssistant data disk); which should appears in the list irrespective to the availability of the NAS.

**Comment by lldaedalusll** [comment]
Same issue here. I can't see any backups in the HA backups page, even though the backup files are there on the CIFS share and get created by schedule.

Log show the timeout messages for me as well:

2023-10-06 12:02:21.474 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /backups/reload request
2023-10-06 12:02:21.474 ERROR (MainThread) [homeassistant.components.hassio] Failed to to call /backups/reload -
2023-10-06 12:04:38.474 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /backups/reload request
2023-10-06 12:04:38.475 ERROR (MainThread) [homeassistant.components.hassio] Failed to to call /backups/reload -
2023-10-06 12:05:02.474 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /backups/reload request
2023-10-06 12:05:02.474 ERROR (MainThread) [homeassistant.components.hassio] Failed to to call /backups/reload -
2023-10-06 12:06:32.475 ERROR (MainThread) [homeassistant.components.hassio.handler] Timeout on /backups/reload request
2023-10-06 12:06:32.476 ERROR (MainThread) [homeassistant.components.hassio] Failed to to call /backups/reload -

Home assistant : 2023.10.0 on rPi4
NAS : CIFS Share on Strato Hidrive

**Comment by jmealo** [comment]
These are all related I think: https://github.com/home-assistant/supervisor/issues/4577 https://github.com/home-assistant/supervisor/issues/4357 https://github.com/home-assistant/supervisor/issues/4358

**Comment by PV-Joe** [comment]
Same Problem here.

**Comment by elboletaire** [comment]
This is still an issue with latest versions.

**Comment by perjury** [comment]
still an issue for me too

**Comment by gzxto1** [comment]
I've started today using SMB for Backups. I ran into the same issue, but found an easy workouround which works for me. On empty backups menu I select "Upload Backups" then the windows file explorer opens, then i close it again without selecting any thing, 5 sec later my 40 backups appear in the list.

**Comment by elboletaire** [comment]
> I've started today using SMB for Backups. I ran into the same issue, but found an easy workouround which works for me. On empty backups menu I select "Upload Backups" then the windows file explorer opens, then i close it again without selecting any thing, 5 sec later my 40 backups appear in the list.

Thanks for the workaround. Unfortunately does not work for me. I still see it empty.

Not sure how is it working for you anyway, since I don't see any calls whatsoever in the websockets when doing that flow. The only way I see any websocket call is by using the reload option (or the first time I access the section). Ofc it never gets the expected response... all the responses I get after a reload are just from sensors and other HA related calls, nothing related to the backups.

**Comment by jmealo** [comment]
This is still an issue. Not stale.

**Comment by PV-Joe** [comment]
Still can‘t see my backups.
Hope thy will fix it soon.

**Comment by jonnybergdahl** [comment]
Please fix this.

**Comment by jhemak** [comment]
+1

**Comment by ToXinE** [comment]
This is still an issue with latest versions.

**Comment by jmealo** [comment]
Just placing some links to relevant source code in case I or someone else picks this up. I asked again on Discord and no contributors have shown any interest.

https://github.com/home-assistant/core/blob/7fbfd446369b8ff4526a92a2018eb00da278e383/homeassistant/components/backup/manager.py#L47C2-L48C57

```python
  async def async_pre_backup(self, hass: HomeAssistant) -> None:
        """Perform operations before a backup starts."""
```

**Comment by jmealo** [comment]
Architecture discussion on a proposed fixed to the Network Storage issues:
https://github.com/home-assistant/architecture/discussions/1033

**Comment by ToXinE** [comment]
still an issue when lots of backups are present in the remote directory

**Comment by FlorianWilk** [comment]
Same issue here. Wasn't a problem in the past, but since a few days. guess @ToXinE idea might be valid.

**Comment by PV-Joe** [comment]
Haven‘t tested this myself yet but maybe it is better to use the advanced AUTO BACKUP integration from HACS to solve the problem?

https://jcwillox.github.io/hass-auto-backup/

**Comment by FlorianWilk** [comment]
> Haven‘t tested this myself yet but maybe it is better to use the advanced AUTO BACKUP integration from HACS to solve the problem?
> 
> https://jcwillox.github.io/hass-auto-backup/

Yes, i do automatic backups - but using a custom script instead of the above integration. But in case you want to restore your backup, or just get an overview of existing backups, the list is useful.
Of course i could get an overview by looking at the NAS-Folder and restore could be done by deleteing/moving every other backupfile except the one you want to restore. But i think it would be much more comfortable to have the existing backup-list working as it should (and does, if you have < 15-20 files)

**Comment by elboletaire** [comment]
> Haven‘t tested this myself yet but maybe it is better to use the advanced AUTO BACKUP integration from HACS to solve the problem?
> 
> [jcwillox.github.io/hass-auto-backup](https://jcwillox.github.io/hass-auto-backup/)

That's what I use, and as said by Florian in the previous message, doees not help/affect to the issue.

**Comment by jonnybergdahl** [comment]
Listing the files should be a very fast operation, but problem occurs because the backend needs to open each and every tar file and read the details to be able to return a complete list to the GUI.
I am not familiar with how the communication between the GUI and the backdoor works, but returning data line by line might be a solution. Or return a list of files, and return detail data as it arrives.

**Comment by QTIPNL** [comment]
don't now if it helps. I had similar kind of symptom as described above. 
I noticed a lot of network traffic on my NAS when opening the backup-screen in Home Assistant for a couple of seconds but not a single backup was shown on the screen even not after a couple of minutes. 
even switching the backup location to local was not showing any result. 
The network drive contained 170+ backups from all sizes. When (re)moving the tar files to another location on the NAS the previous created local files backup files instantly appeared on the backup page of home assistant and creating a newly backup on the network drive also appeared instantly.

**Comment by PV-Joe** [comment]
The problem still persists.
Come on guys 😄

**Comment by ledow** [comment]
I have this same problem, it renders backups invisible to users and makes recovery far more difficult that necessary.

Can we not at least cache backups we know we made, and then test if they're still present only when they're accessed?

**Comment by jhemak** [comment]
Please keep open ... a fix would be most appreciated

**Comment by elboletaire** [comment]
> Please keep open ... a fix would be most appreciated

This issue feels like a clash with @github-actions. I hope we'll win in the end though.

**Comment by agners** [comment]
Do you all have the `Timeout on /backups/reload request` entry in the Core logs as the original poster?

**Comment by bigbear3001** [comment]
@agners i do not have those errors (or any relevant info regarding the backup list request) in the logs.

As i origininally posted in https://github.com/home-assistant/supervisor/issues/4473 (but that got closed before any action was taken from homeassistant) i worked arround the issue by having a second NIC (WIFI + Ethernet) connected to the same network. So for me the first request for listing the backups fails with no backups found in the list and a few seconds later i get the full list of backups on the network storage. If only one NIC is connected to the network i do not get any backups.

**Comment by iMelsom** [comment]
> Do you all have the `Timeout on /backups/reload request` entry in the Core logs as the original poster?

I get this in my core logs: 
Logger: homeassistant.components.hassio
Source: components/hassio/websocket_api.py:135
integration: Home Assistant Supervisor (documentation, issues)
First occurred: 08:20:05 (1 occurrences)
Last logged: 08:20:05

Failed to to call /backups/reload -

While my supervisor logs simply states that: 
2024-06-12 08:19:54.561 INFO (MainThread) [supervisor.backups.manager] Found 3 backup files

No backups are shown...

But during creation of a backup it ususally throws some errors in the superviser log 
(please see attached file: [supervisor_2024-06-12T06-28-39.543Z.log](https://github.com/user-attachments/files/15799447/supervisor_2024-06-12T06-28-39.543Z.log) )
They may be unrelated though..

**Comment by cooki371** [comment]
I got the same issue and I can also show some backups, when moving the rest into a sub directory.

My Supervisor log also contains a bit of information I couldn't find in this issue so far. I am not sure if it is related to the issue itself, but it appears between the logs of some unreadable backup files (NAS connection broke down) and the "Found n backup files" log.

```
[31m2024-07-02 16:32:02.182 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-716' coro=<WSClient.start_listener() done, defined at /usr/src/supervisor/supervisor/homeassistant/websocket.py:98> exception=ConnectionResetError('Cannot write to closing transport')>
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/homeassistant/websocket.py", line 105, in start_listener
    await self._receive_json()
  File "/usr/src/supervisor/supervisor/homeassistant/websocket.py", line 114, in _receive_json
    msg = await self._client.receive()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_ws.py", line 280, in receive
    await self.pong(msg.data)
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_ws.py", line 160, in pong
    await self._writer.pong(message)
  File "/usr/local/lib/python3.12/site-packages/aiohttp/http_websocket.py", line 709, in pong
    await self._send_frame(message, WSMsgType.PONG)
  File "/usr/local/lib/python3.12/site-packages/aiohttp/http_websocket.py", line 675, in _send_frame
    self._write(header + mask + message)
  File "/usr/local/lib/python3.12/site-packages/aiohttp/http_websocket.py", line 702, in _write
    raise ConnectionResetError("Cannot write to closing transport")
ConnectionResetError: Cannot write to closing transport
```

**Comment by drstil** [comment]
Same issue and logs as in description with Core 2024.7.0 and HA OS 12.4. -> The request timed out -> how do increase the request duration (only for this request)?

**Comment by Levante75** [comment]
> still an issue for me too

**Comment by revampatelier** [comment]
+1
Big issue for me too I now have 19 BU's in place across both locations local SSD and NAS but the nas has only 4 while the remaining 15 are on the SSD, I can no longer list or see my backup list so can't restore via my latest BU which resides on the NAS.
I have a connection to a Samba share.

**Comment by Levante75** [comment]
Sure it is a problem. But in the backup you can with the 3 dos upload a 
bu from any location by browsing to it


------ Oorspronkelijke bericht ------
Van "revampatelier" ***@***.***>
Aan "home-assistant/supervisor" ***@***.***>
CC "Robert" ***@***.***>; "Comment" 
***@***.***>
Datum 31-07-2024 20:40:37
Onderwerp Re: [home-assistant/supervisor] Backup page listing empty when 
network storage is configured (Issue #4357)

**Comment by revampatelier** [comment]
Ok..
So I managed to map a windows network drive like so..
\\x.x.x.x\share
I use the share name only not the full path,,
clicked finish and it promted auth details user name and pw.
I  deleted all unnecessary BU's of of the NAS location and voila.. My list appeared as per normal iterating both the SSD and NAS BU's.
So as long as it does not exceed network time out or latency or the number BU's  ( below 19 BU's ) it should list your BU's

**Comment by Mikrofarad** [comment]
From my supervisor log:
> 2024-08-14 08:34:51.619 ERROR (MainThread) [supervisor.backups.backup] Can't read backup tarfile /data/mounts/NAS_Backup/f830cf2d.tar: "filename './backup.json' not found"

I "fixed" the empty list by removing all backup files (*.tar) from the NAS where the backup.json was missing. After that all my 22 backup files where shown in the list (and I have to wait at least 20 seconds).

**Comment by carsten-h** [comment]
I am storing the backup also via SMB on my QNAP NAS. My Home Assistant OS is running on a proxmox server.
Until today the backups are always shown, but starting from today the page shows nothing.
I cleared the browser cache, used differnet Home Assistant Companion apps on iOS/iPadOS/macOS and Android, nothing can be seen.

I see that the backups are created by the daily automation and also when I am creating a backup by hand.

But the page stays empty. I only have seven backups on the NAS.

Home Assistant OS: 13.0
Core: 2024.8.1
Supervisor: 2024.08.0

**Comment by agners** [comment]
@carsten-h your case is most likely related to a HAOS 13.0 issue with Proxmox, see https://github.com/home-assistant/operating-system/issues/3532. Try revert to HAOS 12.4 and see if the problem is gone (`ha os update --version 12.4`).

**Comment by carsten-h** [comment]
> Try revert to HAOS 12.4 and see if the problem is gone (`ha os update --version 12.4`).

Yes, that is working! Thank you!
Then I will wait for a new Home Assistant OS.

**Comment by TeoTN** [comment]
I just started encountering this issue. I first saw it on HA OS v12.4. I updated it minutes ago to v13.1 and the issue persists.
On a side note, it was always a rather poor UX, showing "You don't have any backups yet." for a minute or so when in fact the backups list was loading in the background, now I see that it also shows up when it in fact errors out. The connection is over SMB to a Synology NAS.

### Version set
```
Core 2024.8.2
Supervisor 2024.08.0
Operating System 13.1 (Previously on 12.4)
Frontend 20240809.0
```

### Logs

#### Web browser
<img width="465" alt="Screenshot 2024-08-23 at 21 56 59" src="https://github.com/user-attachments/assets/00f65ff0-681c-4eaa-8606-f98378fb0339">

#### Home Assistant Core
<img width="693" alt="Screenshot 2024-08-23 at 21 58 15" src="https://github.com/user-attachments/assets/e80e6ef2-f837-4155-ac9b-221b6d0072c6">

### Troubleshooting steps
* I tried restarting Home Assistant
* I tried clearing storage in the browser, performing hard refresh + re-logging in
* I ensured IP is correct and provided credentials once again
* I verified that backups are still present on the NAS and the NAS is otherwise reachable
* I upgraded Home Assistant OS from v12.4 to 13.1

**Comment by PV-Joe** [comment]
People put up with it and just move or delete old backups as a workaround.

**Comment by TeoTN** [comment]
I question if that's a viable long-term solution but nonetheless I just moved 50% of my backups to a completely different directory (effectively removing them from the shared space) and it's still not working.

I think (remote) backups must've been broken in one of the recent updates.

**Comment by PV-Joe** [comment]
> I just moved 50% of my backups to a completely different directory (effectively removing them from the shared space) and it's still not working.

Try to leave just 10 backups.
I think it works with up to 30 and that's about it.
It works for me for sure.

**Comment by TeoTN** [comment]
Yeah, going down to 20 "did the thing", so to say. I'm beyond sure though something must've been effed up, a month or so ago it worked flawlessly with 100+ backups for me.

To add insult to injury, I wasn't able to tell which ones I'm (re)moving - I sorted them by modified date and left only ones that were modified in the last 2 months, but apparently this has left me with some old backups from a year ago or older. The backup files have just hash + `.tar` as their names.

All this plus absolute zero problem acknowledgement from any maintainers (the issue keeps self-closing etc) makes it a really sad state of things.

**Comment by Egglestron** [comment]
Still an issue.

**Comment by Otiel** [comment]
Still an issue

**Comment by TeoTN** [comment]
Can confirm, still an issue

**Comment by revampatelier** [comment]
+1 - still an issue for me too..

**Comment by Levante75** [comment]
+1 - still an issue for me too..

**Comment by TeoTN** [comment]
Update: issue persists on version 2024.11.2 running on top of OS v13.2.

**Comment by mbkdev** [comment]
There is an information about the loading of the Page. But its loading about more than half an hour.

This is new for me.

**Comment by TeoTN** [comment]
It seems indeed to finally have some sort of loading spinner but no trace of handling more backups. Version 2024.12.1

**Comment by elboletaire** [comment]
> It seems indeed to finally have some sort of loading spinner but no trace of handling more backups. Version 2024.12.1

I actually noticed an improvement since I couldn’t even list a couple of backups before, and now it’s listing around 60 backups.

So, for me, right now it seems like it's been fixed. I'll update in case there are changes.

**Comment by elboletaire** [comment]
And version 2025.1 finally seems to fix this while adding lots more features to the backups 🥳

**Comment by mbkdev** [comment]
Yes! I have more than 150 Backups and a big file size. I'm very happy to see this working!
But there is one thing: My HA isn't running in English. Everything is in my setted language, but not the backup-page.

**Comment by Levante75** [comment]
Solved core >2024.12


------ Oorspronkelijke bericht ------
Van "github-actions[bot]" ***@***.***>
Aan "home-assistant/supervisor" ***@***.***>
CC "Robert" ***@***.***>; "Comment" 
***@***.***>
Datum 4/02/2025 17:04:55
Onderwerp Re: [home-assistant/supervisor] Backup page listing empty when 
network storage is configured (Issue #4357)

**Comment by MartinHjelmare** [comment]
Let's close this. Mutlitple users are reporting it solved.

There were lots of backup changes in core 2025.1 and there will be more changes in 2025.2. Please open a new issue if there are still problems, one issue per problem after the release tomorrow.
